### DISCUSSION TITLE
SSH access to tulip

### DISCUSSION THREAD

**Discussion by jobf** [shorepine/tulipcc#313] [Ideas]
Is it possible to get SSH running on the tulip? I imagine with it being a micropython environment there may be some limitations here but some way of running remote calls would be super nice.

The main reason I'm asking is that I have the USB host socket in use and desire a way to interact without a keyboard.

**Comment by bwhitman** [comment]
We don't have SSH running (yet!) I believe it would be possible.  But, in the short term, you can just use the USB-UART port (the charge port) to type into / see Tulip's REPL. You will want to add `tulip.tfb_log_start()` to your `boot.py` to show the REPL on the UART, but once you do you can type into it / see it. Connect your computer to it at 115200 baud.

**Reply by jobf** [reply]
Fantastic! Just tested sending commands via the USB-UART port and it works great.

**Reply by CelloGit** [reply]
`tulip.tfb_log_start() `gives error:
`AttributeError: 'module' object has no attribute 'tfb_log_start'`

**Reply by bwhitman** [reply]
Oh yes, since we wrote that we now have `tfb_log_start` on all the time, no need to put that in your boot.py. Also, try `mpremote`! It may be exactly what you need. https://github.com/shorepine/tulipcc/blob/main/docs/getting_started.md#using-mpremote
